.github/workflows/ci.yaml

```diff
@@ -41,26 +41,38 @@ jobs:
     steps:
       - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4
       - name: Get changed files
+        id: changed-files
         uses: tj-actions/changed-files@2d756ea4c53f7f6b397767d8723b3a10a9f35bf2 # v44
         with:
           # globs copied from default settings for run-ormolu
           files: |
             **/*.hs
             **/*.hs-boot
+          # allows the `separator` to actually be a newline (see tj-actions/changed-files#2148)
+          safe_output: false
           separator: "\n"
+      # The job doesn't have permission to push back to contributor forks on contributor PRs, or on protected branches,
+      # so in those cases, we only check the formatting.
       - uses: haskell-actions/run-ormolu@15b0083a0ef416915994fb511652b187f6026a40 # v15
+        if: |
+          github.ref_protected
+            || !(github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name)
+        with:
+          version: ${{ env.ormolu_version }}
+          mode: check
+          pattern: ${{ steps.changed-files.outputs.all_changed_files }}
+      - uses: haskell-actions/run-ormolu@15b0083a0ef416915994fb511652b187f6026a40 # v15
+        if: |
+          !github.ref_protected
+            && github.event.pull_request.base.repo.full_name == github.event.pull_request.head.repo.full_name
         with:
           version: ${{ env.ormolu_version }}
           mode: inplace
           pattern: ${{ steps.changed-files.outputs.all_changed_files }}
       - name: apply formatting changes
-        uses: stefanzweifel/git-auto-commit-action@e348103e9026cc0eee72ae06630dbe30c8bf7a79
-        # v5
-        # and not on a protected branch.
-        # The job doesn't have permission to push back to contributor forks on contributor PRs.
+        uses: stefanzweifel/git-auto-commit-action@e348103e9026cc0eee72ae06630dbe30c8bf7a79 # v5
         if: |
-          always()
```

Comment:
I can't remember why this was here, but we may discover a reason later. It probably belongs somewhere else if anywhere,

Reply:
Yeah, I couldn’t figure out why it would be useful to run this step if an earlier one failed.